yarn-project/p2p/src/client/p2p_client.ts

```diff
@@ -332,7 +332,18 @@ export class P2PClient<T extends P2PClientType = P2PClientType.Full>
    * Uses the batched Request Response protocol to request a set of transactions from the network.
    */
   public async requestTxsByHash(txHashes: TxHash[]): Promise<(Tx | undefined)[]> {
-    const txs = await this.p2pService.sendBatchRequest(ReqRespSubProtocol.TX, txHashes);
+    // Set collective timeout based on the number of txs we are requesting
+    const timeoutMs = Math.max(5000, txHashes.length * 100);
```

Comment:
I'd put a hard cap on this one.

Reply:
We also need to ensure this is capped by the validation reexecution deadline, but that's another story.

Reply:
For re-ex deadline, if it times out as it is unable to get all of the txs, then it will not be able to attest anyway no?